File: android/src/main/kotlin/com/usercentrics/sdk/flutter/bridge/GetUserSessionDataBridge.kt
package com.usercentrics.sdk.flutter.bridge

import com.usercentrics.sdk.flutter.api.FlutterMethodCall
import com.usercentrics.sdk.flutter.api.FlutterResult
import com.usercentrics.sdk.flutter.api.UsercentricsProxy
import com.usercentrics.sdk.flutter.api.UsercentricsProxySingleton

internal class GetUserSessionDataBridge(
    private val usercentrics: UsercentricsProxy = UsercentricsProxySingleton
) : MethodBridge {

    override val name: String
        get() = "getUserSessionData"

    override fun invoke(call: FlutterMethodCall, result: FlutterResult) {
        assert(name == call.method)
        result.success(usercentrics.instance.getUserSessionData())
    }
}

File: ios/Classes/Bridge/GetUserSessionDataBridge.swift
import Usercentrics
import Flutter

struct GetUserSessionDataBridge : MethodBridge {

    let name: String = "getUserSessionData"
    let usercentrics: UsercentricsProxyProtocol

    func invoke(_ call: FlutterMethodCall, _ result: @escaping FlutterResult) {
        assert(call.method == name)
        result(usercentrics.shared.getUserSessionData())
    }
}
